packages/api/src/server/application/service/application.factory.ts

```diff
@@ -396,17 +401,39 @@ export class ApplicationFactory {
   }
 
   // @internal It'll check that the repository exists or not
-  async validateGitProps(gitRequestDTO: GitRequestDTO) {
-    const gitUrl = `${gitRequestDTO.repoUrl}/tree/${gitRequestDTO.gitRef}/${gitRequestDTO.contextDir}`;
-    try {
-      const response = await this.httpService.axiosRef.head(gitUrl);
-      if (response.status === 200) return true;
-    } catch (error) {
-      this.logger.error('GitSource', error);
+  async validateGitProps(repoUrl: string, gitRef: string, contextDir: string) {
+    const gitUrl = this.generateGitUrl(repoUrl, gitRef, contextDir);
+    if (gitUrl.startsWith('https://gitlab')) {
+      try {
+        const response = await this.httpService.axiosRef.get(gitUrl);
+        // @internal for the valid repository Gitlab sends the list of valid urls
+        const regex = /\bhttps?:\/\/[^\s,"}]+\b/g;
+        const gitlabSource = response.data.match(regex).includes(gitUrl.replace(/\/$/, ''));
+        if (gitlabSource && response.status === 200) return true;
+      } catch (error) {
+        this.logger.error('GitlabSource', error);
+      }
+    } else {
+      try {
+        const githubSource = await this.httpService.axiosRef.head(gitUrl);
+        if (githubSource.status === 200) return true;
+      } catch (error) {
+        this.logger.error('GithubSource', error);
+      }
     }
     return false;
   }
 
+  // @internal It'll generate the url for github and gitlab
+  private generateGitUrl(repoUrl: string, gitRef: string, contextDir: string) {
+    let gitUrl;
+    repoUrl = this.getRepoUrl(repoUrl);
+    contextDir = this.getPath(contextDir);
+    if (repoUrl.startsWith('https://github.com')) gitUrl = `${repoUrl}/tree/${gitRef}${contextDir}`;
+    if (repoUrl.startsWith('https://gitlab')) gitUrl = `${repoUrl}/-/tree/${gitRef}${contextDir}`;
+    return gitUrl;
+  }
+
   private getSSRAccessUrl(application: Application, baseUrl: string): string {
```

Comment:
SSR is a technique, do not use it in the method name, 
Is it generating an access URL? If yes, what's the purpose based on those answers? Refactor the method name>
I think the name of this method should be buildAppAccesUrl ?